### ISSUE TITLE
Broken Control Sequences in Reply

### ISSUE THREAD

**Issue by joncatanio** [rampatra/jbot#126]
I'm trying to send back a reply with the encoded user ID so Slack can convert it to the appropriate handle for me. [Here](https://github.com/rampatra/jbot/blob/master/jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java#L162) the message is encoded and the `<` and `>` characters are escaped, preventing the use of [Slack Control Sequences](https://api.slack.com/docs/message-formatting#linking_to_channels_and_users).

Do you provide a way to reply without encoding the message?

#### Example:
I'd like to reply with a string like "Hey what's up <@SLACK_ID>" which will show up in slack like "Hey what's up @handle", where `SLACK_ID` and `handle` would be a user's slack ID and handle respectively.

**Comment by rampatra** [comment]
Thanks for reporting. Will work on it asap.

**Comment by bondj** [comment]
@joncatanio If it helps, I worked around this limitation by adding a method to my bot like this..

	private void replyNoEscape(WebSocketSession session, Event event, Message reply) {
		try {
			if (StringUtils.isEmpty(reply.getType())) {
				reply.setType(EventType.MESSAGE.name().toLowerCase());
			}

			if (reply.getChannel() == null && event.getChannelId() != null) {
				reply.setChannel(event.getChannelId());
			}
			session.sendMessage(new TextMessage(reply.toJSONString()));

		} catch (IOException e) {
			throw new RuntimeException(e);
		}
	}

**Comment by joncatanio** [comment]
Yep! I ended up doing the same thing @bondj, I should've added the same comment for others. Thanks for that!

**Comment by rampatra** [comment]
hey guys, after looking into it, I didn't get the issue completely. Slack suggests us to encode the string before sending it to Slack so why there is a problem if we encode and send?

**Comment by joncatanio** [comment]
The Slack documentation is somewhat misleading on this to be honest. My understanding is that if someone were writing a bot with non-deterministic messages coming from either users or emails or something, then the text should be encoded.

In our case, we're crafting messages that leverage Slack's control sequences, specifically `<` & `>`. In my specific case, my bot will respond to a user and tag them, and the best way to do this is with Slack's control sequences. So my bot will send back an _unencoded_ message `"Hi <@USER_ID>"`, which Slack will convert to `"Hi @joncatanio"`. The issue with the current flow is that the encoded response gets rendered in Slack as `"Hi &lt;@USER_ID&gt;"`, which doesn't make use of the control sequences.

**Comment by joncatanio** [comment]
I believe the best solution might be to simply provide another function that doesn't do the encoding and ultimately let the users decide if they want to send and encoded message (if their use case depends on it) or an unencoded message (allowing for control sequences)?

I think the first sentence of this [documentation](https://api.slack.com/docs/message-formatting#linking_to_channels_and_users) gives the best insight on the use here.

> For example, to refer to a channel or user within a message, **a client should send the following messages**...

I think the keyword here is *client*. JBot can provide methods to send an encoded/unencoded message which depends on someone's usecase.
